### PR TITLE
Javascript javascript1 week2

### PR DESCRIPTION
**Pull request by natata08**
All tasks are done in separate files. Link to my FreeCodeCamp profile in readme.

### CODE REVIEW

File: javascript/javascript1/week2/fullname.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Flight booking fullname function
+
+function getFullname(firstname, surname, useFormalName = false) {
+  if (firstname && surname) {
```
Comment (islanderupnorth) [inline javascript/javascript1/week2/fullname.js]: Remember to write the Javascript names in camelCase. (Even though I know the homework wrote it incorrectly!)

`getFullName`
`firstName`

File: javascript/javascript1/week2/fullname.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    const fullname = `${firstname} ${surname}`;
+    const fullnameFormal = useFormalName ? `${title} ${fullname}` : fullname;
```
Comment (islanderupnorth) [inline javascript/javascript1/week2/fullname.js]: Great use of both template literals but even more so the ternary operator!

File: javascript/javascript1/week2/fullname.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    return fullnameFormal;
+  } else {
+    alert('Please fill in the missing information: First Name and Surname are required.');
+  }
+}
+
+const fullname1 = getFullname('Tom', 'Jones', true);
+const fullname2 = getFullname('', 'Smith', false);
```
Comment (islanderupnorth) [inline javascript/javascript1/week2/fullname.js]: As above `fullName1` but I think you get the idea!

File: javascript/javascript1/week2/fullname.js [hunk 4]
```diff
@@ -1,17 +1,25 @@
 // Flight booking fullname function
 
-function getFullname(firstname, surname, useFormalName = false) {
+function getFullname(firstname, surname, useFormalName = false, isWoman = false) {
```
Comment (islanderupnorth) [inline javascript/javascript1/week2/fullname.js]: Great, setting defaults ✅

File: javascript/javascript1/week2/fullname.js [hunk 5]
```diff
@@ -1,17 +1,25 @@
...
     const fullname = `${firstname} ${surname}`;
-    const fullnameFormal = useFormalName ? `${title} ${fullname}` : fullname;
-    return fullnameFormal;
+    if (!useFormalName) {
+      return fullname;
+    } else if (isWoman) {
+      return `${femaleTitle} ${fullname}`;
+    } else {
+      return `${maleTitle} ${fullname}`;
+    }
```
Comment (islanderupnorth) [inline javascript/javascript1/week2/fullname.js]: You could consider using the ternary operator here:

```
..else {
  return `${isWoman ? femaleTitle : maleTitle} ${fullname}`
}
```

Here I wouldn't say either is necessarily better as it comes down to preference which one thinks is more eligible.

File: javascript/javascript1/week2/event.js [hunk 6]
```diff
@@ -1,9 +1,15 @@
...
 function getEventWeekday(daysUntilEvent) {
-  const weekDays = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
+  const weekdays = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
   const today = new Date();
-  const eventDay = new Date(today.setDate(today.getDate() + daysUntilEvent));
-  const weekDayNumber = eventDay.getDay();
-  return weekDays[weekDayNumber];
+  const todayIndex = today.getDay(); //returns the day of the week (0/sunday to 6/saturday) of a date
+  const eventDayIndex = (todayIndex + daysUntilEvent) % 7;
+  return weekdays[eventDayIndex];
```
Comment (islanderupnorth) [inline javascript/javascript1/week2/event.js:8]: Great job naming conventions and using the modulo operator correctly 🙌

**Review by islanderupnorth — COMMENTED** [review]
Good logic taking the conditions from lowest to highest. Only minor item I can say that is missing is to call the function and log out the result.

**Review by islanderupnorth — COMMENTED** [review]
Works as desired - good job 👍

File: javascript/javascript1/week2/candy-helper.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    i++;
+  }
+  const answer = totalPrice < amountToSpend; //boolean
+  if (answer) {
+    console.log('You can buy more, so please do!');
+  } else {
+    console.log('Enough candy for you!');
+  }
+  return answer;
+}
```
Comment (islanderupnorth) [inline javascript/javascript1/week2/candy-helper.js:39]: What I' missing is to see the functions being called i.e.:

```
addCandy("sweet", 20);
addCandy("chewing-gum", 20);
addCandy("toffee", 20);

canBuyMoreCandy();
```

File: javascript/javascript1/week2/candy-helper.js [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+  while (i < boughtCandyPrices.length) {
+    totalPrice += boughtCandyPrices[i];
+    i++;
+  }
+  const answer = totalPrice < amountToSpend; //boolean
+  if (answer) {
+    console.log('You can buy more, so please do!');
+  } else {
+    console.log('Enough candy for you!');
+  }
```
Comment (islanderupnorth) [inline javascript/javascript1/week2/candy-helper.js]: You could also write this as:

```console.log(answer ? "You can buy more, so please do!" : "Enough candy for you!")```

**Review by islanderupnorth — COMMENTED** [review]
Great work Natalia! You seem to have grasped the concepts very well and writing nice clean code already.

Only general improvement I could see was the final calling of the functions you had created. Minor and easy fixes for you!

File: javascript/javascript1/week2/fullname.js (hunk 1 above)
Comment (natata08) [reply javascript/javascript1/week2/fullname.js]: Changed them.

File: javascript/javascript1/week2/fullname.js (hunk 5 above)
Comment (natata08) [reply javascript/javascript1/week2/fullname.js]: I like it, thank you!

File: javascript/javascript1/week2/candy-helper.js (hunk 7 above)
Comment (natata08) [reply javascript/javascript1/week2/candy-helper.js:39]: Done it!

File: javascript/javascript1/week2/candy-helper.js (hunk 8 above)
Comment (natata08) [reply javascript/javascript1/week2/candy-helper.js]: Done it, thank you!

**Comment by natata08** [comment]
@islanderupnorth thank you so much for your feedback!

**Pull request merged**